packages/flame/lib/src/geometry/shape.dart

```diff
@@ -9,34 +10,125 @@ import 'shape_intersections.dart' as intersection_system;
+  /// Should be the center of that [offsetPosition] and [relativeOffset]
+  /// should be calculated from, if they are not set this is the center of the
+  /// shape
+  Vector2 position = Vector2.zero();
 
   /// The size is the bounding box of the [Shape]
-  Vector2? size;
+  Vector2 size;
+
+  Vector2 get halfSize {
+    if (!_halfSizeCache.isCacheValid([size])) {
+      _halfSizeCache.updateCache(size / 2, [size.clone()]);
+    }
+    return _halfSizeCache.value!;
+  }
 
   /// The angle of the shape from its initial definition
   double angle;
 
-  Vector2 get shapeCenter => position;
+  /// The local position of your shape, so the diff from the [position] of the
+  /// shape
+  Vector2 offsetPosition = Vector2.zero();
+
+  /// The position of your shape in relation to its size from (-1,-1) to (1,1)
+  Vector2 relativeOffset = Vector2.zero();
+
+  /// The [relativeOffset] converted to a length vector
+  Vector2 get relativePosition => (size / 2)..multiply(relativeOffset);
+
+  /// The angle of the parent that has to be taken into consideration for some
+  /// applications of [Shape], for example [HitboxShape]
+  double parentAngle;
+
+  /// The center position of the shape within itself, without rotation
+  Vector2 get localCenter {
+    final stateValues = [
+      size,
+      relativeOffset,
+      offsetPosition,
+    ];
+    if (!_localCenterCache.isCacheValid(stateValues)) {
+      final center = (size / 2)..add(relativePosition)..add(offsetPosition);
+      _localCenterCache.updateCache(
+        center,
+        stateValues.map((e) => e.clone()).toList(growable: false),
+      );
+    }
+    return _localCenterCache.value!;
+  }
 
-  Vector2? _anchorPosition;
-  Vector2 get anchorPosition => _anchorPosition ?? position;
-  set anchorPosition(Vector2 position) => _anchorPosition = position;
+  /// The shape's absolute center with rotation taken into account
+  Vector2 get absoluteCenter {
+    final stateValues = [
+      position,
+      offsetPosition,
+      relativeOffset,
+      angle,
+      parentAngle,
+    ];
+    if (!_absoluteCenterCache.isCacheValid(stateValues)) {
+      /// The center of the shape, before any rotation
+      final center = position + offsetPosition;
+      if (!relativeOffset.isZero()) {
+        center.add(relativePosition);
+      }
+      if (angle != 0 || parentAngle != 0) {
+        center.rotate(parentAngle + angle, center: position);
+      }
+      _absoluteCenterCache.updateCache(center, [
+        position.clone(),
+        offsetPosition.clone(),
+        relativeOffset.clone(),
+        angle,
+        parentAngle,
+      ]);
+    }
+    return _absoluteCenterCache.value!;
+  }
+
+  Vector2 get renderCenter {
+    final zoom = isTranslated ? 1.0 : (camera?.zoom ?? 1);
+    return (isTranslated ? localCenter : absoluteCenter) / zoom;
+  }
+
+  /// Whether the canvas has been translated or not when it comes to rendering
+  /// the shape
+  final bool isTranslated = false;
```

Comment:
should the Shape have to know about the camera? can't we just leave that to its owner?

Reply:
It doesn't have to have an owner, you can render shapes independently too.